app/code/Magento/Catalog/Ui/DataProvider/Product/Listing/Collector/Image.php

```diff
@@ -136,7 +136,6 @@ public function collect(ProductInterface $product, ProductRenderInterface $produ
     public function emulateImageCreating(ProductInterface $product, $imageCode, $storeId, ImageInterface $image)
     {
         $this->storeManager->setCurrentStore($storeId);
-        $this->design->setDefaultDesignTheme();
```

Comment:
I assume that setting the default theme here was made intentionally. Thus after you change this will change the behavior. Maybe it makes sense to move setting the default theme to `collect` method and then restore the original  theme after that?

Reply:
@slavvka yes, I agree. I've been trying to wrap my head around why there is a need change the current theme to the default one since the default one will be set anyhow. But I guess to maintain functionality it's best to fix this more defensively. I'll push the updated changes as per your suggestion.